src/main/java/com/kat/bookstore/service/category/implementation/CategoryServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.kat.bookstore.service.category.implementation;
+
+import com.kat.bookstore.dto.category.CategoryDto;
+import com.kat.bookstore.dto.category.CreateCategoryRequestDto;
+import com.kat.bookstore.entity.category.Category;
+import com.kat.bookstore.exception.EntityNotFoundException;
+import com.kat.bookstore.mapper.category.CategoryMapper;
+import com.kat.bookstore.repository.category.CategoryRepository;
+import com.kat.bookstore.service.category.CategoryService;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto).collect(Collectors.toList());
```

Comment:
Do not ignore pageable object, you can use it:
```suggestion
Page<Category> page = categoryRepository.findAll(pageable);
return page.stream()
  .map(mapper::toDto)
  .toList();
  // or return `Page<CategoryDto>` if you want metadata
```